tree/ntuple/v7/src/RNTupleModel.cxx

```diff
@@ -77,6 +77,35 @@ std::shared_ptr<ROOT::Experimental::RCollectionNTupleWriter> ROOT::Experimental:
    return collectionNTuple;
 }
 
+ROOT::Experimental::Detail::RFieldBase *ROOT::Experimental::RNTupleModel::GetField(std::string_view fieldName)
+{
+   auto split_on_dot = [](std::string_view in) {
+      std::vector<std::string_view> splits;
+      for (std::size_t pos = in.find('.'); pos != std::string_view::npos; pos = in.find('.')) {
+         splits.push_back(in.substr(0, pos));
+         in.remove_prefix(pos + 1);
+      }
+      if (in.size() > 0)
+         splits.emplace_back(in);
+
+      return splits;
+   };
+
+   auto *field = static_cast<ROOT::Experimental::Detail::RFieldBase *>(fFieldZero.get());
+   for (auto subfieldName : split_on_dot(fieldName)) {
+      const auto it =
+         std::find_if(field->begin(), field->end(), [&](const auto &f) { return f.GetName() == subfieldName; });
```

Comment:
The field iterator walk the entire subtree.  Since you are already splitting the levels, I think you are rather looking for `RField::GetSubFields()`.

Reply:
oh, that needs docs :) uhm but then can I not just find the field I want with a single iterator walk?

Reply:
fixed, force-pushed